src/basics.md

```diff
@@ -1101,6 +1101,68 @@ fn main() {
 }
 ```
 
+[ex-error-chain-backtrace]: #ex-error-chain-backtrace
+<a name="ex-error-chain-backtrace"></a>
+## Obtain backtrace of complex error scenarios
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Extend errors by appending new errors via [`chain_err`]. It provides a
+better context to understand an error in a form of a backtrace. The
+below recipes attemps to deserialize the value `256` into a `u8`. An
+error will bubble up from Serde then csv and finaly up to the user
+code.
+
+```rust
+# extern crate csv;
+#[macro_use]
+extern crate error_chain;
+# #[macro_use]
+# extern crate serde_derive;
+#
+# error_chain! {
+#     foreign_links {
+#         Reader(csv::Error);
+#     }
+# }
+
+#[derive(Debug, Deserialize)]
+struct Rgb {
+    red: u8,
+    blue: u8,
+    green: u8,
+}
+
+fn read_rgb_from_csv(csv_data: &[u8]) -> Result<Rgb> {
+    let color: Rgb = csv::Reader::from_reader(csv_data)
+        .deserialize()
+        .nth(0)
+        .ok_or("Cannot deserialize the first CSV record")?
+        .chain_err(|| "Cannot deserialize RGB color")?;
+
+    Ok(color)
+}
+
+# type Hex = u32;
```

Comment:
we can substitute this fun tion with custom `Display` impl for `Rgb`

Reply:
I don't  understand you comment. Can you give more details?

Reply:
> custom Display impl for Rgb
 
Ah, I think I get it :-) Do you mean something like:

```rust
let color = Rgb{...};
println!("#{:X}", color); // Prints #66FFCC
```

`rgb_to_hex` will be replace by the `Display` impl for `Rgb`?

Reply:
Good point :+1: